keep/conditions/condition_factory.py

```diff
@@ -5,9 +5,11 @@
 
 class ConditionFactory:
     @staticmethod
-    def get_condition(condition_type, condition_config) -> BaseCondition:
+    def get_condition(
+        condition_type, condition_name, condition_config
```

Comment:
why was condition_name added? is it so we can access the condition "results"?

Reply:
because the previous structure was:
- steps
- - conditions results

so steps.bla.conditions[0] was the first condition

the problem is that in "foreach" mode, there is more than one condition result per condition so the structure does not apply anymore, so now its
- steps
- - conditions
- - - results